### PR TITLE
Support remote endpoint.

### PR DESCRIPTION
**Pull request by artursouza**
# Description

Support remote sidecar configuration via consistent env variable.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: https://github.com/dapr/java-sdk/issues/876

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: sdk/src/test/java/io/dapr/client/DaprClientGrpcTest.java [hunk 1]
```diff
@@ -91,52 +92,39 @@ public class DaprClientGrpcTest {
...
-          }
-        } catch (IOException e) {
-        }
-      });
-      t.start();
-      client.waitForSidecar(10000).block();
-    }
+  public void waitForSidecarOK() {
+    Mockito.doReturn(Mono.error(new RuntimeException())).when(channel).waitForChannelReady(10000);
+    assertThrows(RuntimeException.class, () -> client.waitForSidecar(10000).block());
```
Comment (mukundansundar) [inline sdk/src/test/java/io/dapr/client/DaprClientGrpcTest.java:127]: this is also testing for exception. 
should be changed

File: sdk-tests/src/test/java/io/dapr/it/DaprRun.java [hunk 2]
```diff
@@ -140,6 +140,10 @@ public void use() {
     System.getProperties().setProperty(
         Properties.API_METHOD_INVOCATION_PROTOCOL.getName(),
         DaprApiProtocol.GRPC.name());
+    System.getProperties().setProperty(
+            Properties.GRPC_ENDPOINT.getName(), "http://127.0.0.1:" + this.ports.getGrpcPort());
```
[the remark this answers is not in the document]
Comment (artursouza) [reply sdk-tests/src/test/java/io/dapr/it/DaprRun.java:144]: It is an endpoint, so the schema is part of it. Users will also use http vs https and the port will be parsed automatically.

File: sdk/src/test/java/io/dapr/client/DaprClientGrpcTest.java (hunk 1 above)
Comment (artursouza) [reply sdk/src/test/java/io/dapr/client/DaprClientGrpcTest.java:127]: Bad copy and paste. Fixing it.

File: sdk/src/main/java/io/dapr/client/DaprClientBuilder.java [hunk 3]
```diff
@@ -162,19 +163,28 @@ private DaprClient buildDaprClient(DaprApiProtocol protocol) {
...
+  private ManagedChannel buildGrpcManagedChanel() {
+    String host = Properties.SIDECAR_IP.get();
     int port = Properties.GRPC_PORT.get();
-    if (port <= 0) {
-      throw new IllegalArgumentException("Invalid port.");
+    boolean insecure = true;
+    String grpcEndpoint = Properties.GRPC_ENDPOINT.get();
+    if ((grpcEndpoint != null) && !grpcEndpoint.isEmpty()) {
+      URI uri = URI.create(grpcEndpoint);
+      insecure = uri.getScheme().equalsIgnoreCase("http");
```
Comment (artursouza) [inline sdk/src/main/java/io/dapr/client/DaprClientBuilder.java:179]: See, we use the schema here.

File: sdk-tests/src/test/java/io/dapr/it/DaprRun.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (mukundansundar) [reply sdk-tests/src/test/java/io/dapr/it/DaprRun.java:144]: why are we using a http scheme for a gRPC endpoint?

[the remark this answers is not in the document]
Comment (artursouza) [reply sdk-tests/src/test/java/io/dapr/it/DaprRun.java:144]: gRPC can operate over HTTP. In this case, we use to parse the port. The .NET SDK for gRPC communication takes in the entire URL as an endpoint. See https://learn.microsoft.com/en-us/aspnet/core/tutorials/grpc/grpc-start?view=aspnetcore-7.0&tabs=visual-studio#create-the-greeter-client

**Review by mukundansundar — APPROVED** [review]

**Pull request merged**